src/utils/filter.ts

```diff
@@ -0,0 +1,25 @@
+import { FilteredBy } from '../types/FilterBy';
+import { Todo } from '../types/Todo';
+
+export function getFilteredTodos(
+  todos: Todo[],
+  { filterBy, query }: { filterBy: FilteredBy, query: string },
+) {
+  let preparedTodos = [...todos];
+
+  if (filterBy === 'active') {
+    preparedTodos = todos.filter(todo => !todo.completed);
+  }
+
+  if (filterBy === 'completed') {
+    preparedTodos = todos.filter(todo => todo.completed);
+  }
+
+  if (query) {
+    return preparedTodos.filter(todo => todo.title
+      .toLowerCase()
+      .includes(query.toLowerCase()));
+  }
```

Comment:
don't forget to trim your search value
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/54936044/780ee74b-f0d5-43e5-a246-c577b0153c69)